crates/ruff_db/src/vendored.rs

```diff
@@ -0,0 +1,379 @@
+use std::cell::{RefCell, RefMut};
+use std::io::{self, Read};
+use std::ops::{Deref, DerefMut};
+use std::sync::{Mutex, MutexGuard};
+
+use itertools::Itertools;
+use once_cell::sync::Lazy;
+use zip::{read::ZipFile, ZipArchive, ZipWriter};
+
+pub use path::{VendoredPath, VendoredPathBuf};
+
+pub mod path;
+
+type Result<T> = io::Result<T>;
+
+#[derive(Debug)]
+pub struct VendoredFileSystem {
+    inner: VendoredFileSystemInner,
+}
+
+impl Default for VendoredFileSystem {
+    fn default() -> Self {
+        static DATA: Lazy<Box<[u8]>> = Lazy::new(|| {
+            let mut buffer = Vec::new();
+            let cursor = io::Cursor::new(&mut buffer);
+            {
+                let mut archive = ZipWriter::new(cursor);
+                archive.finish().unwrap();
+            }
```

Comment:
What's the use case for implementing `Default`? I think  I strongly prefer not to implement `Default` for this type because it just isn't very useful when empty. 

If you want to support an empty `VendoredFS`, then I think I would model it as an enum with `EMPTY` and `Data(&'static [u8]`) variants.

Reply:
I implemented `Default` here because you'd implemented `Default` on the `VendoredVfs` enum, and I assumed you had a reason for that 😆

I don't see a great reason for implementing `Default` on this struct; I can get rid of it.